parser/src/string.rs

```diff
@@ -472,12 +470,13 @@ impl<'a> StringParser<'a> {
                         }
                     }
                     if !content.is_empty() {
+                        let range = self.current_range(location).sub_end(1.into());
```

Comment:
What's the reason for substracting one character here? Is it safe to assume that the preceding character is ASCII? Should we instead pass the `fstring_start_location` as a function argument?